python/ray/_private/function_manager.py

```diff
@@ -227,61 +243,55 @@ def get_execution_info(self, job_id, function_descriptor):
         Returns:
             A FunctionExecutionInfo object.
         """
-        if self._worker.load_code_from_local:
-            # Load function from local code.
-            # Currently, we don't support isolating code by jobs,
-            # thus always set job ID to NIL here.
-            job_id = ray.JobID.nil()
+
+        try:
+            # Try to load function from local code first.
             if not function_descriptor.is_actor_method():
-                self._load_function_from_local(job_id, function_descriptor)
-        else:
+                self._load_function_from_local(function_descriptor)
+        except:
             # Load function from GCS.
             # Wait until the function to be executed has actually been
             # registered on this worker. We will push warnings to the user if
             # we spend too long in this loop.
             # The driver function may not be found in sys.path. Try to load
             # the function from GCS.
             with profiling.profile("wait_for_function"):
-                self._wait_for_function(function_descriptor, job_id)
+                self._wait_for_function(function_descriptor,job_id)
+
         try:
             function_id = function_descriptor.function_id
-            info = self._function_execution_info[job_id][function_id]
+            info = self._function_execution_info[function_id]
         except KeyError as e:
             message = ("Error occurs in get_execution_info: "
                        "job_id: %s, function_descriptor: %s. Message: %s" %
                        (job_id, function_descriptor, e))
             raise KeyError(message)
         return info
 
-    def _load_function_from_local(self, job_id, function_descriptor):
+    def _load_function_from_local(self, function_descriptor):
```

Comment:
This method should better just return a boolean. The exception is no longer useful.